React/Modules/RCTEventEmitter.m

```diff
@@ -78,21 +78,22 @@ - (void)dealloc
     RCTLogError(@"`%@` is not a supported event type for %@. Supported events are: `%@`",
                 eventName, [self class], [[self supportedEvents] componentsJoinedByString:@"`, `"]);
   }
-  if (_listenerCount == 0) {
+  _listenerCount++;
+  if (_listenerCount == 1) {
     [self startObserving];
   }
-  _listenerCount++;
 }
 
 RCT_EXPORT_METHOD(removeListeners:(NSInteger)count)
 {
   if (RCT_DEBUG && count > _listenerCount) {
     RCTLogError(@"Attempted to remove more %@ listeners than added", [self class]);
   }
-  if (count == _listenerCount) {
+  _listenerCount -= count;
+  _listenerCount = (_listenerCount <= 0) ? 0 : _listenerCount;
```

Comment:
do I need to import anything? fairly new to objective-c :smile: I usually work on the Android side of things